app/models/user.rb

```diff
@@ -1,5 +1,5 @@
 class User < ApplicationRecord
-  has_one :authentication_provider_github, dependent: :destroy
+  has_one :authentication_provider_github, require: true, dependent: :destroy
```

Comment:
To represent User and AuthenticationProviderGithub is always one-to-one pair,
this adds `require: true` option to the `has_one` association between them.

Reply:
Withdrawing. This change causes small troubles in the rspec. Finally I added small type narrowing check to the main code.